CautionaryAlertsApi/V1/Gateways/UhGateway.cs

```diff
@@ -131,6 +133,26 @@ public async Task<IEnumerable<CautionaryAlertListItem>> GetCautionaryAlertsByMMH
             return alerts.Select(x => x.ToDomain());
         }
 
+        public CautionaryAlert GetCautionaryAlertByAlertId(AlertQueryObject query)
+        {
+            var alerts = _uhContext.PropertyAlertsNew
+                        .Where(x => x.MMHID == query.PersonId.ToString())
+                        .Where(x => x.AlertId == query.AlertId.ToString())
+                        .ToList();
+
+            if (alerts.Count() == 0) return null;
+
+            var cautionaryAlert = alerts.Select(x => x.ToCautionaryAlertDomain());
+
+            //We should never expect the count to by more than one as AlertId should be unique but adding this condition as we only return the first alert. 
+            if (cautionaryAlert.Count() > 1)
+                throw new MoreThanOneAlertException(cautionaryAlert.Count());
+
+
+            return cautionaryAlert.First();
```

Comment:
This should be `.FirstOrDefault()` in case your query has no values.
So you can return a .FirstOrDefault() which remove the need for line 143-145

```
if (cautionaryAlert.Count() > 1)
                throw new MoreThanOneAlertException(cautionaryAlert.Count());
                
return cautionaryAlert.FirstOrDefault();
```


And for max efficiency, it should done without a `.ToList()` upstream.